src/Glpi/Assets/ImportMapGenerator.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Assets;
+
+use Glpi\Application\Environment;
+use Plugin;
+use Psr\SimpleCache\CacheInterface;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use Symfony\Component\Filesystem\Path;
+
+/**
+ * Generates an import map for JavaScript modules with cache busting parameters
+ *
+ * @final
+ */
+class ImportMapGenerator
+{
+    /**
+     * @var string
+     */
+    private $root_doc;
+
+    /**
+     * @var string
+     */
+    private $glpi_root;
+
+    /**
+     * @var CacheInterface|null
+     */
+    private $cache;
+
+    /**
+     * @param string $root_doc Root document URL path
+     * @param CacheInterface|null $cache Optional cache instance
+     */
+    public function __construct(string $root_doc, string $glpi_root, ?CacheInterface $cache = null)
+    {
+        $this->root_doc  = $root_doc;
+        $this->glpi_root = $glpi_root;
+        $this->cache     = $cache;
+    }
+
+    /**
+     * Get the list of active plugins
+     *
+     * @return array Array of plugin names
+     */
+    protected function getPluginDirList(): array
+    {
+        return array_map(
+            fn($plugin_key) => Plugin::getPhpDir($plugin_key, true),
+            Plugin::getPlugins()
+        );
+    }
+
+    /**
+     * Generate the import map data
+     *
+     * @return array{imports: array<string, string>} The import map data with module names as keys and URLs as values
+     */
+    public function generate(): array
+    {
+        $should_use_cache = $this->cache !== null && !Environment::get()->shouldExpectResourcesToChange();
```

Comment:
You should split the cache. One entry for GLPI, and one entry for each plugin. The cache for the corresponding plugin should be reset when a plugin is updated (the key must be added in `Plugin::resetHookableCacheEntries()`).